File: gaia-sdk-javascript/src/graphql/request/input/DeleteApiKeyImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to delete an api key instance
 */
export class DeleteApiKeyImpulse {
public _typeName = "DeleteApiKeyImpulse";
    private apiKeyId:Uuid;

    constructor (apiKeyId:Uuid) {
        this.apiKeyId = apiKeyId;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/DeleteApiKeyImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class DeleteApiKeyImpulse():
    """
    The specification to delete an api key instance
    """
    apiKeyId: str

    def __init__(self, apiKeyId: str):
        self.apiKeyId = apiKeyId

    def __eq__(self, other):
        if type(other) is type(self):
            return self.apiKeyId == other.apiKeyId
        return False

    def __repr__(self):
        return {'apiKeyId': self.apiKeyId}
